widgets/secretProviders/src/RemoveSecretProviderButton.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react';
+import { tableRefreshEvent } from './SecretProvidersTable.consts';
+import { RequestStatus } from './types';
+import type { SecretProvidersWidget } from './widget.types';
+
+const { Icon, Confirm: DeleteModal } = Stage.Basic;
+
+const { useBoolean } = Stage.Hooks;
+
+const t = Stage.Utils.getT(`widgets.secretProviders`);
+
+interface RemoveSecretProviderButtonProps {
+    secretProvider: SecretProvidersWidget.DataItem;
+    toolbox: Stage.Types.Toolbox;
+}
+
+const RemoveSecretProviderButton = ({ secretProvider, toolbox }: RemoveSecretProviderButtonProps) => {
+    const [isModalVisible, showModal, hideModal] = useBoolean();
+    const [deletingStatus, setDeletingStatus] = useState<RequestStatus>(RequestStatus.INITIAL);
+    const deleteModalContent = t('deleteModal.content', {
+        secretProviderId: secretProvider.id
+    });
+
+    const removeSecretProvider = () => {
+        setDeletingStatus(RequestStatus.SUBMITTING);
+
+        toolbox
+            .getManager()
+            .doDelete(`/secrets-providers/${secretProvider.id}`)
+            .then(() => {
+                setDeletingStatus(RequestStatus.SUBMITTED);
+                toolbox.getEventBus().trigger(tableRefreshEvent);
+            })
+            .catch(() => {
+                setDeletingStatus(RequestStatus.ERROR);
+            });
```

Comment:
I'm not sure we need `RequestStatus`.

What do you think about:
```suggestion
            .then(() => toolbox.getEventBus().trigger(tableRefreshEvent))
            .catch(() => setDeletingStatus(RequestStatus.ERROR))
            .finally(hideModal);
```
?

Will it work the same or I'm missing something? 🤔